### PR TITLE
docs: pull beats doc changes into APM

### PR DESCRIPTION
**Pull request by bmorelli25**
## Motivation/summary

This PR syncs APM documentation with Beats documentation.

### CODE REVIEW

File: docs/copied-from-beats/docs/template-config.asciidoc [hunk 1]
```diff
@@ -23,9 +23,14 @@ endif::[]
 You can adjust the following settings to load your own template or overwrite an
 existing one.
 
-*`setup.template.enabled`*:: Set to false to disable template loading. If set this to false,
+*`setup.template.enabled`*:: Set to false to disable template loading. If this is set to false,
 you must <<load-template-manually,load the template manually>>.
 
+*`setup.template.type`*:: The type of template to use. Available options: `legacy` (default), index templates
+before Elasticsearch v7.8. Use this to avoid breaking existing deployments. New options are `component`
```
Comment (jalvz) [inline docs/copied-from-beats/docs/template-config.asciidoc:30]: @axw is this true also for us?

Comment (axw) [reply docs/copied-from-beats/docs/template-config.asciidoc:30]: Good catch. No, we should not document this for apm-server.

**Review by axw — CHANGES_REQUESTED** [review]

File: docs/copied-from-beats/docs/template-config.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/copied-from-beats/docs/template-config.asciidoc:30]: Thank you! Will get this fixed.

Comment (bmorelli25) [reply docs/copied-from-beats/docs/template-config.asciidoc:31]: Fixed in c6c931ba4b0dcc05b0ddb4589467df54b76197b1.

**Comment by bmorelli25** [comment]
jenkins retest this please

**Review by axw — APPROVED** [review]
Thanks!

**Review by jalvz — APPROVED** [review]

**Pull request merged**